### PR TITLE
Deprecate '--cluster-xx' options and add warning

### PR DESCRIPTION
**Pull request by aiordache**
**- What I did**
Deprecated `--cluster-advertise`, `--cluster-store` , `--cluster-store-opt` flags from dockerd CLI.
 - logged warning when used

**- How to verify it**
```
$ dockerd --cluster-advertise 192.168.1.5:2376 --cluster-store etcd://192.168.1.6:2379
Flag --cluster-advertise has been deprecated, Deprecated option.
Flag --cluster-store has been deprecated, Deprecated option.
WARN[2020-02-12T17:43:14.108302242Z] The "cluster-advertise" option is deprecated. To be removed soon. 
WARN[2020-02-12T17:43:14.108351655Z] The "cluster-store" option is deprecated. To be removed soon. 
...
```
```
$ cat /etc/docker/daemon.json 
{
	"cluster-store": "etcd://192.168.1.5:1257",
	"cluster-store-opts": {},
	"cluster-advertise": "192.168.1.6:3123"
}
```

```
$ dockerd
WARN[2020-02-12T17:47:20.767700485Z] The "cluster-advertise" option is deprecated. To be removed soon. 
WARN[2020-02-12T17:47:20.767745964Z] The "cluster-store" option is deprecated. To be removed soon. 
INFO[2020-02-12T17:47:20.767757515Z] Starting up                                  
INFO[2020-02-12T17:47:20.768774398Z] libcontainerd: started new containerd process  pid=582
INFO[2020-02-12T17:47:20.768816420Z] parsed scheme: "unix"                         module=grpc
...
```

**- A picture of a cute animal (not mandatory but encouraged)**

Co-authored-by: Yves Brissaud <yves.brissaud@gmail.com>

Signed-off-by: Anca Iordache <anca.iordache@docker.com>

relates to https://github.com/moby/moby/issues/40383

### CODE REVIEW

**Comment by thaJeztah** [comment]
Windows failure is unrelated (see the discussion on https://github.com/moby/moby/pull/40512#issuecomment-585748486)

File: cmd/dockerd/config.go [hunk 1]
```diff
@@ -68,9 +68,14 @@ func installCommonConfigFlags(conf *config.Config, flags *pflag.FlagSet) error {
 	flags.Var(opts.NewNamedListOptsRef("labels", &conf.Labels, opts.ValidateLabel), "label", "Set key=value labels to the daemon")
 	flags.StringVar(&conf.LogConfig.Type, "log-driver", "json-file", "Default driver for container logs")
 	flags.Var(opts.NewNamedMapOpts("log-opts", conf.LogConfig.Config, nil), "log-opt", "Default log driver options for containers")
+
 	flags.StringVar(&conf.ClusterAdvertise, "cluster-advertise", "", "Address or interface name to advertise")
+	_ = flags.MarkDeprecated("cluster-advertise", "Deprecated option.")
```
Comment (AkihiroSuda) [inline cmd/dockerd/config.go]: Can we add a more descriptive text as in `flags.MarkDeprecated("restart", "Please use a restart policy on docker run")`? 

Maybe like `flags.MarkDeprecated("cluster-advertise", "Swarm classic is deprecated. Please use Swarm-mode (docker swarm init)")`?

File: cmd/dockerd/daemon.go [hunk 2]
```diff
@@ -469,6 +470,18 @@ func loadDaemonCliConfig(opts *daemonOptions) (*config.Config, error) {
 	return conf, nil
 }
 
+func warnOnDeprecatedConfigOptions(config *config.Config) {
```
Comment (AkihiroSuda) [inline cmd/dockerd/daemon.go]: This function should also handle  `--restart`?

**Comment by AkihiroSuda** [comment]
LGTM but a couple of nits

File: cmd/dockerd/daemon.go (hunk 2 above)
Comment (aiordache) [reply cmd/dockerd/daemon.go]: It is mostly intended for the options set in the daemon configuration file. Can't find if --restart has an equivalent in the daemon config file.

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Pull request merged**